core/src/components/input/input.md.outline.scss

```diff
@@ -189,6 +192,19 @@
   width: calc(var(--padding-start) - #{$input-md-floating-label-padding});
 }
 
+/**
+ * If a large --border-radius is applied through CSS, the
+ * .input-outline-start element will not be wide enough with
+ * default padding, causing the left and right sides of the
+ * border to show different radii. Increasing the width of
+ * this element fixes the issue, but this can't be done with
+ * stacked/floating labels because it pushes the notch spacer
+ * to the side, breaking the visuals when the label floats up.
+ */
+:host(.input-fill-outline:not(.input-label-placement-stacked):not(.input-label-placement-floating)) .input-outline-start {
```

Comment:
Out of curiosity, why was having developers use `--padding-start` not used as a solution? For example, developers customizing border radius already have to do this:

```css
ion-input.custom-border-radius-input {
  --border-radius: 32px;
}
```

Since they are already in the styles, it doesn't seem like much additional effort to do the following:

```css
ion-input.custom-border-radius-input {
  --border-radius: 32px;
  --padding-start: 32px;
}
```

The benefit of this is that it keeps the solution consistent regardless of `fill`. The proposed PR would make it so developers would need to understand implementation details of `fill` in order to know if they need `--padding-start` to make the border radius work as expected.

Additionally, very specific selectors like the one here can make it challenging for users to override since they need to beat the specificity of this selector. Components like `ion-input` use scoped styles which already presents some challenges for customizing re: specificity: https://ionicframework.com/docs/api/input#css-custom-properties

Demo: https://codepen.io/liamdebeasi/pen/abRKywR

------

I took a look at other libraries that implement the notch effect, and the results are similar to the defect ours has.

Material Design Components for the Web: https://material-components.github.io/material-components-web-catalog/#/component/text-field

![image](https://github.com/ionic-team/ionic-framework/assets/2721089/5c967ba3-4540-4a40-ab58-50decb46ed10)

MUI: https://mui.com/material-ui/react-text-field/

![image](https://github.com/ionic-team/ionic-framework/assets/2721089/0fb3e3fb-9734-4fbe-8925-683e144e7114)

MUI avoids part of this issue by using a [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset), but the padding of the floating label is still incorrect. Notice how it gets cut off at an awkward angle: 
![image](https://github.com/ionic-team/ionic-framework/assets/2721089/22428ff5-cc10-4687-bf74-4d3373772f1a)

Reply:
I'm okay with just having people do that, honestly. I figured it we can fix it for some cases, it couldn't hurt, but you make valid points. I'd like to keep these commits in case we need the fix in the future; does closing this PR and making a new one with the fix added to the `--border-radius` descriptions sound good? (Or do you think it's better placed somewhere else?)

Reply:
Yeah I think that makes sense 👍